packages/block-library/src/gallery/gap-styles.js

```diff
@@ -1,15 +1,24 @@
 /**
  * WordPress dependencies
  */
-import { BlockList } from '@wordpress/block-editor';
+import { BlockList, store as blockEditorStore } from '@wordpress/block-editor';
 import { useContext, createPortal } from '@wordpress/element';
+import { useSelect } from '@wordpress/data';
 
 export default function GapStyles( { blockGap, clientId } ) {
+	const themeSupportsLayout = useSelect( ( select ) => {
+		const { getSettings } = select( blockEditorStore );
+		return getSettings()?.supportsLayout;
+	}, [] );
 	const styleElement = useContext( BlockList.__unstableElementContext );
 
+	const gapFallback = themeSupportsLayout ? '0.5em' : '16px';
+	const backwardsCompatGap = ! themeSupportsLayout
+		? `; gap: ${ gapFallback }`
+		: '';
 	const gap = blockGap
 		? `#block-${ clientId } { --wp--style--unstable-gallery-gap: ${ blockGap } }`
-		: `#block-${ clientId } { --wp--style--unstable-gallery-gap: var( --wp--style--block-gap, 0.5em ) }`;
+		: `#block-${ clientId } { --wp--style--unstable-gallery-gap: var( --wp--style--block-gap, ${ gapFallback } ) ${ backwardsCompatGap }  }`;
```

Comment:
Should we add some inline comments about the different values and also about `backwardsCompatGap` to help our future selves?

I don't know why the following reads better for me. Please ignore. There's no real saving in line numbers or anything useful 😄 

```js
const gapFallbackValue = themeSupportsLayout ? '0.5em' : '16px';
const blockGapValue = blockGap || `var( --wp--style--block-gap, ${ gapFallback } )`

const gapCSS = `
#block-${ clientId } { 
    --wp--style--unstable-gallery-gap: ${ blockGap };  
    { ! themeSupportsLayout ? 'gap: 16px;' : '' }
}`;


```